chroma_core/models/stratagem.py

```diff
@@ -0,0 +1,399 @@
+    StorageResourceRecord,
+)
+
+
+class StratagemConfiguration(StatefulObject):
+    filesystem_id = models.IntegerField(
+        help_text="The filesystem id associated with the stratagem configuration", null=False
+    )
+    interval = models.IntegerField(help_text="Interval value in seconds between each stratagem execution", null=False)
+    report_duration = models.IntegerField(
+        help_text="Interval value in seconds between stratagem report execution", null=False
+    )
+    report_duration_active = models.BooleanField(
+        default=False, help_text="Indicates if the report should execute at the given interval"
+    )
+    purge_duration = models.IntegerField(help_text="Interval value in seconds between stratagem purges", null=False)
+    purge_duration_active = models.BooleanField(
+        default=False, help_text="Indicates if the purge should execute at the given interval"
+    )
+
+    states = ["unconfigured", "configured"]
+    initial_state = "unconfigured"
+
+    class Meta:
+        app_label = "chroma_core"
+        ordering = ["id"]
+
+
+class StratagemUnconfiguredAlert(AlertStateBase):
+    default_severity = logging.ERROR
+
+    def alert_message(self):
+        return "Stratagem did not configure correctly"
+
+    class Meta:
+        app_label = "chroma_core"
+
+    def end_event(self):
+        return AlertEvent(
+            message_str="%s started" % self.alert_item,
+            alert_item=self.alert_item.primary_host,
+            alert=self,
+            severity=logging.INFO,
+        )
+
+    def affected_targets(self, affect_target):
+        affect_target(self.alert_item)
+
+
+class ConfigureStratagemTimerStep(Step):
+    def run(self, kwargs):
+        job_log.debug("Create stratagem timer step kwargs: {}".format(kwargs))
+        # Create systemd timer
+
+
+class ConfigureStratagemJob(StateChangeJob):
+    state_transition = StateChangeJob.StateTransition(StratagemConfiguration, "unconfigured", "configured")
+    stateful_object = "stratagem_configuration"
+    stratagem_configuration = models.ForeignKey(StratagemConfiguration)
+
+    display_group = Job.JOB_GROUPS.COMMON
+    display_order = 10
+
+    requires_confirmation = False
+    state_verb = "Configure Stratagem"
+
+    class Meta:
+        app_label = "chroma_core"
+        ordering = ["id"]
+
+    @classmethod
+    def long_description(cls, stateful_object):
+        return help_text["configure_stratagem_long"]
+
+    def description(self):
+        return help_text["configure_stratagem_description"]
+
+    def get_steps(self):
+        steps = [(ConfigureStratagemTimerStep, {})]
+
+        return steps
+
+
+class RunStratagemStep(Step):
+    def run(self, args):
+        host = args["host"]
+        path = args["path"]
+        target_name = args["target_name"]
+        report_duration = args["report_duration"]
+        purge_duration = args["purge_duration"]
+
+        def calc_purge_duration(report_duration, purge_duration):
```

Comment:
Shouldn't this be calculating report duration not purge duration?

Reply:
From our conversation on June 21'st:
```
(atime < sys_time - 5 && atime > sys_time - 10)
```
Maybe we can discuss again?

Reply:
Perhaps I'm mistaken but I think it's the report time that needs the upper bound.

`|--Report--||--Purge---|`

Reply:
correct, i've upated.